.github/workflows/build_and_test_dbt.yaml

```diff
@@ -62,7 +62,7 @@ jobs:
         run: |
           if [[ $MODIFIED_RESOURCES_ONLY == 'true' ]]; then
             echo "Running tests on modified/new resources only"
-            dbt test -t "$TARGET" -s state:modified state:new --state "$STATE_DIR"
+            dbt test -t "$TARGET" -s state:modified state:new --defer --state "$STATE_DIR"
```

Comment:
This fixes a small bug that I hadn't noticed before working on this PR. Prior to this change, the lack of a `--defer` flag meant that tests would fail when run against views or tables that were not built as part of the CI environment (see [here](https://github.com/ccao-data/data-architecture/actions/runs/6041375559/job/16394210971#step:9:1) for a sample failing workflow run). The lack of certain views or tables in a CI environment is intentional: We don't want to build models in CI unless their definition is new or has changed on the PR branch, which helps us speed up CI by only building and testing models that are being affected by the PR. The `--defer` tells dbt that if a model does not exist in the CI version of the DAG, but it _does_ exist in the prod version of the DAG, then it should refer to the prod version for the purpose of running builds and tests. (The exception to this is models that have changed as part of the PR, which are always built in the CI environment due to the `-s state:modified` flag.)

Detailed docs on dbt's deferral system are [here](https://docs.getdbt.com/reference/node-selection/defer).